src/mds/Server.cc

```diff
@@ -5722,6 +5722,44 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::return_bytes(string& value_str)
+{
+  if (value_str.empty()) {
+    return -CEPHFS_EINVAL;
+  }
+  unordered_map<char, int64_t> size_values = {{'k', 1L << 10}, {'m', 1L << 20}, 
+                                             {'g', 1L << 30}, {'t', 1L << 40}};
+  char data_size = tolower(value_str.back());
+  string value = value_str.substr(0, value_str.length() - 1);
+  if (size_values.find(data_size) != size_values.end()) {
+    try {
+      return (boost::lexical_cast<int64_t>(value)) * size_values[data_size];
+    } catch (boost::bad_lexical_cast const&) {
+      if(value.empty()) {
+        dout(10) << "coefficient is empty for '" << data_size 
+        << "' size spec" << dendl;
+      } else {
+        dout(10) << "bad coefficient: '" << value << "' for '" << data_size 
+        << "' size spec" << dendl;
+        }
+      return -CEPHFS_EINVAL;
+    }
+  } else {
+    try {
+      return boost::lexical_cast<int64_t>(value_str);
+    } catch (boost::bad_lexical_cast const&) {
+      if(value.empty()) {
+        dout(10) << "invalid single character alphabetic value: " 
+        << value_str << dendl;
+      } else {
+        dout(10) << "could not identify the data size " 
+        << data_size << " in " << value_str << dendl;
+        }
```

Comment:
nit: message rewording ...
`"invalid size spec: '" << value_str << "'"`

Since the last_char wasn't a valid size unit, we are free to assume that all the chars are numeric, but since the lexical_cast fails, we simply specify that the size spec was invalid. Testing for the coefficient to be an empty string doesn't make sense since the size unit (last_char) wasn't a valid one. So we don't need to specialize the log message in this case.

Reply:
invalid size spec: '100Z' - quotes look overkill here. invalid size spec: 100Z would be better.

Reply:
done.

Reply:
> invalid size spec: '100Z' - quotes look overkill here. invalid size spec: 100Z would be better.

maybe I'm a bit paranoid ... but I was thinking about the value being "100 Z" with a space between 100 and Z :)
and so the quotes to identify the value as a single token

Reply:
I think i would like to keep it simple, `invalid spec: 100Z` is good to go i think.